tests/test_quaternion.py

```diff
@@ -256,6 +256,15 @@ def test_average(self, arr):
         average = Quaternion.average(*q_with_noise, weights=weights)
         assert average == q_with_noise[0] or average == -q_with_noise[0]
 
+        # test empty list averaging
+        try:
+            empty = Quaternion.average(*[])
+            self.fail('Expected QuaternionError exception to be raised')
+        except QuaternionError:
+            pass
+        except np.linalg.LinAlgError:
```

Comment:
Why this exception in particular?
Could it be just `except Exception`?

Reply:
Yes, i guess that's better.

Just to explain why i met this particular exception:
`np.linalg.LinAlgError` is what happens when we try to `np.linalg.eigh(matrix)` for a matrix of rank less than two.

In the case of an empty quaternion list gives us:

```python
b = np.array([])
weights = np.array([])
```
then:
```python
m = 0.0
```

So we end up calling:
```python
cls._first_eigenvector(0.0)
```

which leads to:
```python
np.linalg.eigh(0.0)
--> np.linalg.LinAlgError
```
___

In this test, `except Exception` is much better, since for example, `_first_eigenvector` could change in the future, checking the rank of its input and raising exceptions if they aren't met. That would break this test unnecessarily.